packages/rocketchat-lib/server/lib/sendNotificationsOnMessage.js

```diff
@@ -270,31 +279,32 @@ RocketChat.callbacks.add('afterSaveMessage', function(message, room, userId) {
 			RocketChat.Sandstorm.notify(message, [userOfMention._id],
 				`@${ user.username }: ${ message.msg }`, 'privateMessage');
 
-		}
-		if ((userOfMention != null) && canBeNotified(userOfMentionId, 'mobile')) {
-			const duration = settings.desktopNotificationDurations[userOfMention._id];
-			notifyDesktopUser(userOfMention._id, user, message, room, duration);
-		}
+			if (canBeNotified(userOfMentionId, 'desktop')) {
+				const duration = settings.desktopNotificationDurations[userOfMention._id];
+				notifyDesktopUser(userOfMention._id, user, message, room, duration);
+			}
 
-		if ((userOfMention != null) && canBeNotified(userOfMentionId, 'desktop')) {
-			if (Push.enabled === true && userOfMention.statusConnection !== 'online') {
-				RocketChat.PushNotification.send({
-					roomId: message.rid,
-					username: push_username,
-					message: push_message,
-					badge: getBadgeCount(userOfMention._id),
-					payload: {
-						host: Meteor.absoluteUrl(),
-						rid: message.rid,
-						sender: message.u,
-						type: room.t,
-						name: room.name
-					},
-					usersTo: {
-						userId: userOfMention._id
-					}
-				});
-				return message;
+			if (canBeNotified(userOfMentionId, 'mobile')) {
+				if (Push.enabled === true && userOfMention.statusConnection !== 'online') {
+					RocketChat.PushNotification.send({
+						roomId: message.rid,
+						username: push_username,
+						message: push_message,
+						badge: getBadgeCount(userOfMention._id),
+						payload: {
+							host: Meteor.absoluteUrl(),
+							rid: message.rid,
+							sender: message.u,
+							type: room.t,
+							name: room.name
+						},
+						usersTo: {
+							userId: userOfMention._id
+						},
+						category: canSendMessageToRoom(room, userOfMention.username) ? 'MESSAGE' : 'MESSAGE_NOREPLY'
```

Comment:
I see that you're duplicating the strings... maybe use constants?

Reply:
If everything else is working as planned, I would only change this strings to constants before merging.

Reply:
agreed. I thought doing this, but since we don't have project wide constants (which would make sense for this), I decided using raw strings..

I'll change for in-file constants though. :wink:

Reply:
changed to constants.. can you review again pls?